static/js/components/dashboard/courses/StatusMessages.js

```diff
@@ -133,8 +134,41 @@ export const calculateMessages = (props: CalculateMessagesProps) => {
     if (hasPaidForAnyCourseRun(course)) {
       // exam is required, user has not yet passed it
       if (exams && ! passedExam) {
+        let messageBox = {};
+        if (failedExam) {
+          if (course.can_schedule_exam) { // if can schedule exam now
+            if (course.has_to_pay) {
+              messageBox = {
+                message: "You did not pass the exam. If you want to re-take the exam, you need to pay again.",
+                action: courseAction(firstRun, COURSE_ACTION_PAY)
+              };
+            } else {
+              messageBox['message'] = `You did not pass the exam, but you can try again.
+               Click above to reschedule an exam with Pearson.`;
+            }
+          } else if (R.isEmpty(course.exams_schedulable_in_future)) {
+            // no info about future exam runs
+            messageBox['message'] = `You did not pass the exam. There are currently no exams
+             available for scheduling. Please check back later.`;
+          } else {
+            // can not schedule now, but some time in the future
+            if (course.has_to_pay) {
+              messageBox = {
+                message: `You did not pass the exam. If you want to re-take the exam, you need
+                 to pay again. You can sign up to re-take the exam starting
+                  on ${formatDate(course.exams_schedulable_in_future[0])}`,
```

Comment:
Need to ensure (either in the UI, or in `MMTrack.get_future_exam_runs`) that this list is sorted in ascending chronological order so `course.exams_schedulable_in_future[0]` is deterministically the next run.